File: atcoder.jp/contests/abc286/tasks/abc286_a/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const [_N, P, Q, R, S] = input().split(" ").map(Number);
const A = input().split(" ").map(Number);

const ans = A.slice(0, P - 1)
  .concat(A.slice(R - 1, S))
  .concat(A.slice(Q, R - 1))
  .concat(A.slice(P - 1, Q))
  .concat(A.slice(S));

console.log(ans.join(" "));

File: atcoder.jp/contests/abc286/tasks/abc286_a/main.rs
use itertools::Itertools;
use proconio::input;

fn main() {
    input! {
        n: usize,
        p: usize,
        q: usize,
        r: usize,
        s: usize,
        a: [usize; n],
    }

    let ans = a[..p - 1]
        .iter()
        .chain(&a[r - 1..s])
        .chain(&a[q..r - 1])
        .chain(&a[p - 1..q])
        .chain(&a[s..])
        .join(" ");

    println!("{}", ans);
}
